zhaquirks/niko/switch.py

```diff
@@ -0,0 +1,335 @@
+"""Niko switches (only Niko Double connectable switch,10A (552-721X2) at this time)"""
+
+from typing import Final
+
+from zigpy import types as t
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+)
+from zigpy.zcl.clusters.homeautomation import Diagnostic
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+from zhaquirks import EventableCluster
+from zhaquirks.const import (
+    ARGS,
+    BUTTON_1,
+    BUTTON_2,
+    BUTTON_3,
+    BUTTON_4,
+    COMMAND_CLICK,
+    COMMAND_HOLD,
+    COMMAND_RELEASE,
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.niko import NIKO
+
+
+class ButtonOperationMode(t.enum8):
+    # WARN: while one can technically write 0x00 to the operationMode attribute
+    #       this seems to brick the device and it will need to be rejoined
+    Decoupled = 0x01
+    ControlRelay = 0x02
+
+
+class LedState(t.enum_factory(t.uint8_t)):
+    Off = 0x00
+    On = 0xFF
+
+
+class LedOperationMode(t.enum8):
```

Comment:
Today I did some more testing. ButtonOperation works successfully but LedOperationMode does not seem to make any changes on the device.
@arvidc were you able to test this in your tests from https://github.com/zigpy/zha-device-handlers/issues/2286?
Any pointer how I can debug this?

Reply:
So the attribute isn't written correctly? Writing `0` or `1` doesn't work?
If so, did you try setting the type for the ZCLAttributeDef to t.uint8 and see if it works then?

Reply:
Hey @svenjochems , I've been using it through a custom quirk on my end and it seems to work (apart from the potentially unintuitive behaviour described in the comments; which makes me think `LedOperationMode` is perhaps better called `LedEnabled` with states `Disabled` and `Enabled`, but that's beyond this).
More specifically, I wrote `LedOperationMode` to `0` for most/all(?) of my switches, and actively control `LedState` on one to reflect our terrace light.

I do have them as:
` 
   class LedState(t.uint24_t):
        Off = 0
        On = 255

    class LedOperationMode(t.uint8_t):
        Decoupled = 0
        ControlledByRelay = 1`
        
and I can't succesfully write (and it doesn't read like that either) e.g. `LedOperationMode.Decoupled` (in analogy to `ButtonOperationMode`), but only `0` or `1`. Maybe because this isn't a proper enum, as I believe @TheJulianJES was pointing out earlier?

(Sorry, I barely know what an enum is, let alone a `t.uint24_t`. So pretty much all I do is copy and adapt pieces from somewhere and trial-and-error until I think it works well enough for me.)

Reply:
> Maybe because this isn't a proper enum

Yes, the "Zigbee type" is just an integer. Not an enum. If the zigpy enum is used, zigpy also sends it as a Zigbee enum which we do not want in this case. This might be changed in the future.

For now, just set the `type` for the `ZCLAttributeDef` to `t.uint8_t` (to force zigpy to send it as a Zigbee int).
We can keep the enum for future-use (with quirks v2 for exposing the entity or when this is fixed. But that's what required for now.
Just inheriting `t.uint8_t` in a class "works", but doesn't really do anything different from just using `t.uint8_t` when other class attributes are added.